### PR TITLE
[Runtime] Use substrate 3.0.0

### PR DESCRIPTION
**Pull request by MyronFanQiu**
Bump substrate version to 3.0.0

### CODE REVIEW

File: cstrml/balances/src/benchmarking.rs [hunk 1]
```diff
@@ -1,5 +1,19 @@
-// Copyright (C) 2019-2021 Crust Network Technologies Ltd.
-// This file is part of Crust.
+// This file is part of Substrate.
```
Comment (zikunfan) [inline cstrml/balances/src/benchmarking.rs:1]: Copyright

File: cstrml/balances/src/tests_local.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
-// Copyright (C) 2019-2021 Crust Network Technologies Ltd.
-// This file is part of Crust.
+// This file is part of Substrate.
```
Comment (zikunfan) [inline cstrml/balances/src/tests_local.rs:1]: Copyright

File: cstrml/balances/src/tests_reentrancy.rs [hunk 3]
```diff
@@ -0,0 +1,310 @@
+// This file is part of Substrate.
```
Comment (zikunfan) [inline cstrml/balances/src/tests_reentrancy.rs]: cp

File: cstrml/balances/src/weights.rs [hunk 4]
```diff
@@ -1,6 +1,6 @@
-// Copyright (C) 2019-2021 Crust Network Technologies Ltd.
-// This file is part of Crust.
+// This file is part of Substrate.
```
Comment (zikunfan) [inline cstrml/balances/src/weights.rs]: cp

File: cstrml/market/Cargo.toml [hunk 5]
```diff
@@ -9,44 +9,44 @@ repository = "https://github.com/crustio/crust/"
...
-frame-benchmarking = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false, optional = true }
+frame-benchmarking = { default-features = false, optional = true, version = "3.0.0" }
 
 # crust runtime modules
-primitives = { version = "0.11.1", package = 'cst-primitives', path = '../../primitives', default-features = false }
+primitives = { package = "cst-primitives", path = "../../primitives", default-features = false }
 
 [features]
-default = ['std']
+default = ["std"]
```
Comment (zikunfan) [inline cstrml/market/Cargo.toml:32]: 🤙🏻

File: node/src/service.rs [hunk 6]
```diff
@@ -209,14 +204,13 @@ pub fn new_full(mut config: Configuration) -> Result<TaskManager, ServiceError>
...
         remote_blockchain: None,
-        telemetry_connection_sinks: telemetry_connection_sinks.clone(),
         network_status_sinks,
         system_rpc_tx,
     })?;
 
     let (babe_block_import, grandpa_link, babe_link) = import_setup;
 
-    if is_authority {
+    if let sc_service::config::Role::Authority { .. } = &role {
```
Comment (zikunfan) [inline node/src/service.rs:213]: Change back to `is_authority`

File: node/src/service.rs [hunk 7]
```diff
@@ -291,7 +282,7 @@ pub fn new_full(mut config: Configuration) -> Result<TaskManager, ServiceError>
 
     let grandpa_config = sc_finality_grandpa::Config {
         // FIXME: [Substrate]substrate/issues#1578 make this available through chain spec
-        gossip_duration: Duration::from_millis(1000),
+        gossip_duration: Duration::from_millis(333),
```
Comment (zikunfan) [inline node/src/service.rs:285]: 1000

File: runtime/src/lib.rs [hunk 8]
```diff
@@ -159,6 +159,7 @@ parameter_types! {
 		})
 		.avg_block_initialization(AVERAGE_ON_INITIALIZE_RATIO)
 		.build_or_panic();
+	pub const SS58Prefix: u8 = 66;
```
Comment (zikunfan) [inline runtime/src/lib.rs:162]: Open https://github.com/crustio/crust/issues/506 for differ different chains

Comment (MyronFanQiu) [reply runtime/src/lib.rs:162]: Sure.

**Review by zikunfan — DISMISSED** [review]

File: cstrml/staking/src/lib.rs [hunk 9]
```diff
@@ -1407,7 +1406,8 @@ decl_module! {
         #[weight = T::DbWeight::get().reads_writes(4, 7)
             .saturating_add(76 * WEIGHT_PER_MICROS)]
         fn reap_stash(_origin, stash: T::AccountId) {
-            ensure!(T::Currency::total_balance(&stash).is_zero(), Error::<T>::FundedTarget);
+            let at_minimum = T::Currency::total_balance(&stash) == T::Currency::minimum_balance();
+			ensure!(at_minimum, Error::<T>::FundedTarget);
```
Comment (zikunfan) [inline cstrml/staking/src/lib.rs:1410]: extra tab

Comment (MyronFanQiu) [reply cstrml/staking/src/lib.rs]: good catch. Fixed

**Review by zikunfan — DISMISSED** [review]

**Review by zikunfan — APPROVED** [review]

**Pull request merged**